src/helpers/git_helpers/branch_configuration_helpers.sh

```diff
@@ -104,9 +103,37 @@ function ensure_knows_parent_branches {
       else
         # here we don't know the parent of the current branch -> ask the user
         echo
-        echo -n "Please enter the parent branch for $(echo_inline_cyan_bold "$current_branch") ($(echo_inline_dim "$MAIN_BRANCH_NAME")): "
+        local branches=$(git for-each-ref --sort=-committerdate refs/heads/ --format='%(refname:short)')
+
+        function print_branch {
+          local number=$1
+          local branch_name=$2
+
+          output_style_bold
+          printf "%3s: " "$number"
+          output_style_reset
+          echo "$branch_name"
+          branch_numbers[number]="$branch_name"
+        }
+
+        print_branch 1 "$MAIN_BRANCH_NAME"
+        i=2
+        for branch in $branches; do
+          if [ "$branch" != "$current_branch" -a "$branch" != "$MAIN_BRANCH_NAME" ]; then
+            print_branch $i "$branch"
+            i=$(( i + 1 ))
+          fi
+        done
+
+        echo
+        echo -n "Please enter the parent branch name or number for $(echo_inline_cyan_bold "$current_branch") ($MAIN_BRANCH_NAME): "
```

Comment:
Good feedback! I have improved the wording according to your suggestion, and added the 10-second summary of branch nesting. Please apply the @allewun style and sanity checker over it again!  :)